composer/datasets/ade20k.py

```diff
@@ -388,6 +391,135 @@ def initialize_object(self, batch_size, dataloader_hparams) -> DataSpec:
+    min_resize_scale: float = hp.optional("Minimum value that the image and target can be scaled", default=0.5)
+    max_resize_scale: float = hp.optional("Maximum value that the image and target can be scaled", default=2.0)
+    final_size: int = hp.optional("Final size of the image and target", default=512)
+    ignore_background: bool = hp.optional("If true, ignore the background class in training loss", default=True)
+
+    def validate(self):
+        if self.split not in ['train', 'val', 'test']:
+            raise ValueError(f"split value {self.split} must be one of ['train', 'val', 'test'].")
+
+        if self.base_size <= 0:
+            raise ValueError("base_size cannot be zero or negative.")
+
+        if self.min_resize_scale <= 0:
+            raise ValueError("min_resize_scale cannot be zero or negative")
+
+        if self.max_resize_scale < self.min_resize_scale:
+            raise ValueError("max_resize_scale cannot be less than min_resize_scale")
+
+    def initialize_object(self, batch_size: int, dataloader_hparams: DataLoaderHparams) -> DataSpec:
+        self.validate()
+
+        if self.split == 'train':
+            both_transform = torch.nn.Sequential(
+                RandomResizePair(min_scale=self.min_resize_scale,
+                                 max_scale=self.max_resize_scale,
+                                 base_size=(self.base_size, self.base_size)),
+                RandomCropPair(
+                    crop_size=(self.final_size, self.final_size),
+                    class_max_percent=0.75,
+                    num_retry=10,
+                ),
+                RandomHFlipPair(),
+            )
+
+            # Photometric distoration values come from mmsegmentation:
+            # https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/datasets/pipelines/transforms.py#L837
+            r_mean, g_mean, b_mean = IMAGENET_CHANNEL_MEAN
+            image_transform = torch.nn.Sequential(
+                PhotometricDistoration(brightness=32. / 255, contrast=0.5, saturation=0.5, hue=18. / 255),
+                PadToSize(size=(self.final_size, self.final_size), fill=(int(r_mean), int(g_mean), int(b_mean))))
+
+            annotation_transform = PadToSize(size=(self.final_size, self.final_size), fill=0)
+        else:
+            both_transform = None
+            image_transform = transforms.Resize(size=(self.final_size, self.final_size),
+                                                interpolation=TF.InterpolationMode.BILINEAR)
+            annotation_transform = transforms.Resize(size=(self.final_size, self.final_size),
+                                                     interpolation=TF.InterpolationMode.NEAREST)
+
+        remote = os.path.join(self.remote, self.split)
+        local = os.path.join(self.local, self.split)
+        dataset = StreamingADE20k(remote, local, self.shuffle, both_transform, image_transform, annotation_transform, batch_size)
```

Comment:
@knighton just a heads up, here's where I pass the device `batch_size` to the dataset, so it can coordinate it's workers to split their indices in a way such that there is only 1 incomplete batch per epoch.

Reply:
Ty ty